pkg/kn/commands/plugin/plugin.go

```diff
@@ -53,16 +53,13 @@ func BindPluginsFlagToViper(cmd *cobra.Command) {
 	viper.SetDefault("lookup-plugins", false)
 }
 
-// AllowedExtensibleCommandGroups the list of command groups that can be
-// extended with plugins, e.g., a plugin named `kn-source-kafka` for Kafka
-// event sources is allowed. This is defined as a fixed [...]string since
-// cannot defined Golang []string constants
-var AllowedExtensibleCommandGroups = [...]string{"source"}
+// CoreCommandNames names of all core `kn` commands
+var CoreCommandNames = []string{}
```

Comment:
I don't see the point why to track the list of all commands as you can get the list anyway when checking for the plugin. Can't we just remove `CoreCommanNames` and `InAllowedExtensibleCommandGroups()` as the latter will always return true when you iterate over all commands ?

Reply:
Also it looks to me that "InAllowedExtensibleCommandGroups()" is not really used by any code ?

Reply:
Yes, nice catch. I had `InAllowedExtensibleCommandGroups()`before fore previous version for `source` extension. Not needed anymore. Will remove.

Reply:
Actually, I am using it in the `verifier.go`:

```go
	// Check both, commands with underscore and with dash because plugins can be called with both
	overwrittenCommands := make(map[string]bool)
	for _, c := range [][]string{cmds, convertUnderscoresToDashes(cmds)} {
		cmd, _, err := v.root.Find(c)
		if err == nil {
			if !InAllowedExtensibleCommandGroups(cmd.Name()) {
				overwrittenCommands[cmd.CommandPath()] = true
			}
		}
	}
```

I'll move it there since it's a freeform function and make it private. Initially it was used in multiple places (root.go and verifier.go) and I needed to have it in a place that made sense for both.